ATCA Time Scan

rlodico06 opened the pull request:
Created new versions of the scan.py script for Zones 2 and 3. The Zone 2 version has been tested and functions as intended. The Zone 3 version will be tested when there is a window of availability. The two script versions are identical with the exception of PV names.
<img width="373" alt="Scan Script Results" src="https://github.com/user-attachments/assets/215ffb98-2e42-4df3-ae06-6aedab25092a">

rlodico06 committed:
Zone 2 scan.py config

rlodico06 committed:
Commented Target Time Print

rlodico06 committed:
Flipped scan.py Polarity

rlodico06 committed:
Update scan.py wait time

rlodico06 committed:
Reconfig Zone 2 scan.py

rlodico06 committed:
Increased scan.py wait time

rlodico06 committed:
Merge pull request #37 from slaclab/main

Update Zone 2 Testing Branch

rlodico06 committed:
Updated scan.py for Zone 2 Checkout

rlodico06 committed:
Added Phase Shift Timer

rlodico06 committed:
Power Operator Bug Fix

rlodico06 committed:
Phase Shift Timer Update

rlodico06 committed:
Removed logger feature

rlodico06 committed:
Added phase shifter delay units

rlodico06 committed:
Separate Scripts for Zones 2 and 3

rlodico06 committed:
Added back LCLS-I scan.py

rlodico06 committed:
Wait for ctr to match tgt

The phase shift timer will now wait for the counter time to match the target time before calculating the phase shift delay.

rlodico06 committed:
Added wait for ctr time in Zone 3

rlodico06 committed:
Added wait to phase shift delay test

rlodico06 committed:
Add wait to zone 3 scan.py

rlodico06 commented:
Recommendations from discussion with Dring and Tyler:
- Spit out error instead of target time and counter time
- Query the counter time while looping to check for any unexpected changes in counter time
- Compare on the phase motor status instead of counter
- Add exception handling for large changes in tabletop counter time

junyangx commented:
Compare on phase motor status in addition to

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
From: rlodico06 ***@***.***>
Sent: Tuesday, October 29, 2024 5:06:39 PM
To: slaclab/femto-timing ***@***.***>
Cc: Xiang, Dring ***@***.***>; Review requested ***@***.***>
Subject: Re: [slaclab/femto-timing] ATCA Time Scan (PR #38)


BEWARE: This email originated outside of our organization. DO NOT CLICK links or attachments unless you recognize the sender and know the content is safe.


Recommendations from discussion with Dring and Tyler:

  *   Spit out error instead of target time and counter time
  *   Query the counter time while looping to check for any unexpected changes in counter time
  *   Compare on the phase motor status instead of counter
  *   Add exception handling for large changes in tabletop counter time

—
Reply to this email directly, view it on GitHub<https://github.com/slaclab/femto-timing/pull/38#issuecomment-2445542276>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AYOPIL7YUGQZIHLETJEEF53Z6API7AVCNFSM6AAAAABQ2XNN26VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDINBVGU2DEMRXGY>.
You are receiving this because your review was requested.Message ID: ***@***.***>

slactjohnson commented on exp-timing/zone2_scan.py:
```diff
@@ -0,0 +1,54 @@
+# time scan
+for x in range(0, stop):
+    if (x<(stop/2)):
+        caput(tgt_pv,tgt+init_step*(mult**x), wait=True) # step up with increasing step size
```
This setup is kind of weird. Why are you using `caput` after already creating a `PV` class for the PV? You can just use `tgt_pg.put()`. At the beginning of the script you can instantiate your PV classes and then just use those throughout. 

https://pyepics.github.io/pyepics/pv.html#the-pv-class

slactjohnson reviewed (changes requested):
This needs a little work before I would approve a merge. 
Let me know if you have any questions.

rlodico06 committed:
Scan Revisions Pt. 1

stephf0716 commented:
This is not the right repo to put this script into. This is where the femto script for LCLS-I lives. For shared code for our group, please add it to [tss-source](https://github.com/slaclab/tss-source).
I also agree with Tyler's comments, make the changes requested and open a PR in the other repo with the same reviewers.

rlodico06 committed:
Created a common scan function

I removed code that was redundant between Zones 2 and 3 and created a common scan function.

rlodico06 commented:
I have made a round of changes to address these comments and created [a new PR in the tss-source repo](https://github.com/slaclab/tss-source/pull/2), per Stephanie's request.
